spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/lock/DefaultJdbcClient.java

```diff
@@ -0,0 +1,141 @@
+
+	/**
+	 * Default value for the table prefix property.
+	 */
+	public static final String DEFAULT_TABLE_PREFIX = "INT_";
+
+	private enum Query {
+
+		DELETE("DELETE FROM %SLOCK WHERE REGION=? AND LOCK_KEY=? AND CLIENT_ID=?"),
+
+		DELETE_EXPIRED("DELETE FROM %SLOCK WHERE REGION=? AND LOCK_KEY=? AND CLIENT_ID=? AND CREATED_DATE<?"),
+
+		DELETE_ALL("DELETE FROM %SLOCK WHERE REGION=? AND CLIENT_ID=?"),
+
+		UPDATE("UPDATE %SLOCK SET CREATED_DATE=? WHERE REGION=? AND LOCK_KEY=? AND CLIENT_ID=?"),
+
+		INSERT("INSERT INTO %SLOCK (REGION, LOCK_KEY, CLIENT_ID, CREATED_DATE) VALUES (?, ?, ?, ?)"),
+
+		COUNT("SELECT COUNT(REGION) FROM %SLOCK WHERE REGION=? AND LOCK_KEY=? AND CLIENT_ID=? AND CREATED_DATE>=?");
+
+		private String sql;
+
+		Query(String sql) {
+			this.sql = sql;
+		}
+
+		public String getSql(String prefix) {
+			return String.format(this.sql, prefix);
+		}
+	}
+
+	private String id = UUID.randomUUID().toString();
+
+	private int ttl = 10000;
+
+	private final JdbcTemplate template;
+
+	private volatile String prefix = DEFAULT_TABLE_PREFIX;
+
+	private volatile String region = "DEFAULT";
+
+	@Autowired
+	DefaultJdbcClient(DataSource dataSource) {
+		this.template = new JdbcTemplate(dataSource);
+	}
+
+	/**
+	 * A unique grouping identifier for all locks persisted with this store. Using
+	 * multiple regions allows the store to be partitioned (if necessary) for different
+	 * purposes. Defaults to <code>DEFAULT</code>.
+	 *
+	 * @param region the region name to set
+	 */
+	public void setRegion(String region) {
+		Assert.hasText(region, "Region must not be null or empty.");
+		this.region = region;
+	}
+
+	/**
+	 * @param prefix the prefix to set (default INT_)
+	 */
+	public void setPrefix(String prefix) {
+		this.prefix = prefix;
+	}
+
+	@Override
+	public void close() {
+		this.template.update(Query.DELETE_ALL.getSql(this.prefix), this.region, getId());
+	}
+
+	@Override
+	public void delete(String lock) {
+		this.template.update(Query.DELETE.getSql(this.prefix), this.region, lock, getId());
+	}
+
+	@Transactional(isolation = Isolation.SERIALIZABLE, timeout = 1)
+	@Override
+	public boolean acquire(String lock) {
+		deleteExpired(lock);
+		if (this.template.update(Query.UPDATE.getSql(this.prefix), new Date(), this.region, lock, getId()) > 0) {
+			return true;
+		}
+		try {
+			return this.template.update(Query.INSERT.getSql(this.prefix), this.region, lock, getId(), new Date()) > 0;
+		}
+		catch (DuplicateKeyException e) {
+			return false;
+		}
+	}
+
+	@Override
+	public boolean isAcquired(String lock) {
+		deleteExpired(lock);
+		return this.template.queryForObject(Query.COUNT.getSql(this.prefix), Integer.class, this.region, lock, getId(),
+				new Date(System.currentTimeMillis() - this.ttl)) == 1;
+	}
+
+	private int deleteExpired(String lock) {
+		return this.template.update(Query.DELETE_EXPIRED.getSql(this.prefix), this.region, lock, getId(),
+				new Date(System.currentTimeMillis() - this.ttl));
```

Comment:
Actually I'm sure the logic of this `deleteExpired()` method is about to avoid dead lock and let any other client in the distributed system to free the lock and obtain it. Therefore this `DELETE` must be free from `CLIENT_ID` restriction.